release-tools/release_image_github.sh

```diff
@@ -0,0 +1,223 @@
+#!/bin/bash
+# Copyright (c) 2018 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+script_name="${0##*/}"
+script_dir=$(dirname $(realpath -s "$0"))
+
+readonly tmp_dir=$(mktemp -t -d cc-bump.XXXX)
+release_tool="${tmp_dir}/release-tool"
+osbuilder_dir="${tmp_dir}/osbuilder/"
+repo_owner="clearcontainers"
+push=false
+
+versions_file="https://raw.githubusercontent.com/clearcontainers/runtime/master/versions.txt"
+os_version="$(curl -Ls ${versions_file} | grep '^clear_vm_image_version=' | cut -d= -f 2)"
+agent_version="$(curl -Ls ${versions_file} | grep '^cc_agent_version=' | cut -d= -f 2)"
+
+function cleanup {
+	if [ -d "${osbuilder_dir}" ]; then
+		info "Remove temp dir: ${osbuilder_dir}"
+		#osbuilder generated files are own by root
+		sudo rm -rf "${osbuilder_dir}"
+	fi
+
+	rm  -rf "${tmp_dir}"
+}
+
+trap cleanup EXIT
+
+die() {
+	echo >&2 -e "\e[1mERROR\e[0m: $*"
+	exit 1
+}
+
+info() {
+	echo -e "\e[1mINFO\e[0m: $*"
+}
+
+
+
+source "${script_dir}/../versions.txt"
+
+
+build_image() {
+	export USE_DOCKER=1
+
+	AGENT_VERSION="${agent_version}" OS_VERSION="${os_version}" make rootfs
+	make image
+}
+
+build_release_tool() {
+	pushd "${script_dir}/../release-tools/"
+	info "build release tool"
+	go build -o  "${release_tool}"
+	popd
+}
+
+get_image_version() {
+	local agent_version="$1"
+	local os_version="$2"
+
+	[ -n ${agent_version} ] || die "need agent version"
+	[ -n ${os_version} ] || die "need os version"
+
+	echo "cc-${os_version}-agent-${agent_version:0:6}"
+}
+
+get_dist_name(){
+	local image_version=$1
+	[ -n ${image_version} ] || die "need image version"
+	echo "image-${image_version}-binaries"
+}
+
+release_image() {
+	info "Creating image tarball"
+	image_version=$(get_image_version "$agent_version" "$os_version")
+	dist_bin_name=$(get_dist_name "${image_version}")
+	image_name="${image_version}.img"
+	mkdir -p "${dist_bin_name}"
+
+	cp workdir/container.img "${dist_bin_name}/${image_name}"
```

Comment:
- Can you avoid hard-coding `workdir` - where is that coming from btw?
- Also, can you add a variable for `container.img`?

Reply:
that is from osbuilder, because I did not changed the workdir in from osbuilder it is creating the image in the current current working directory, yep I will change make this explicit.

Reply:
Added a variable to make it explicit.